plugin/vcs/gitlab/gitlab.go

```diff
@@ -580,15 +577,16 @@ func refreshToken(instanceURL string, oldToken *string, oauthContext oauthContex
 	if err != nil {
 		return fmt.Errorf("failed to send refresh token POST %v (%w)", url, err)
 	}
-	if err := getOAuthErrorDetails(resp); err != nil {
-		return err
-	}
-
+	// We should not call getOAuthErrorDetails otherwise the reponse body is read and will be empty.
 	body, err = io.ReadAll(resp.Body)
```

Comment:
Since we change the getOAuthErrorDetails return value, we can still call it?

Reply:
Updated the comment. We probably shouldn't determine oauth error inside the function of refreshing oauth token.

Reply:
I guess I still don't fully understand here.

I understand we need to retrieve the body to get the refreshOauthResponse. But that doesn't conflict with using getOAuthErrorDetails

Since we can get the body first and pass it to getOAuthErrorDetails.

getOAuthErrorDetails is still useful since it checks oauth error and here we are fetching the new token using the ouathContext so it's possible that we can get oauth error.

Reply:
I see. So we should update the comment?

Reply:
Done